checker/tests/lock/TestTreeKinds.java

```diff
@@ -331,65 +487,131 @@ public void testMethodAnnotations() {
 
   if (r.nextBoolean()) {
     lockTheLock();
-    // Yes, lock reassignments are bad, but for the purposes of this
-    // experiment they are useful.
-    @LockHeld ReentrantLock heldLock = lock;
-    foo = new Object();
+    foo.field.toString();
 
     unlockTheLock();
 
-    //:: error: (assignment.type.incompatible)
-    @LockHeld ReentrantLock heldLock2 = lock;
     //:: error: (contracts.precondition.not.satisfied.field)
-    foo = new Object();
+    foo.field.toString();
   }
   else {
-    //:: error: (assignment.type.incompatible)
-    @LockHeld ReentrantLock heldLock = lock;
     //:: error: (contracts.precondition.not.satisfied.field)
-    foo = new Object();
+    foo.field.toString();
   }
 
   if (tryToLockTheLock()) {
-    @LockHeld ReentrantLock heldLock = lock;
-    foo = new Object();
+    foo.field.toString();
   }
   else {
-    //:: error: (assignment.type.incompatible)
-    @LockHeld ReentrantLock heldLock = lock;
     //:: error: (contracts.precondition.not.satisfied.field)
-    foo = new Object();
+    foo.field.toString();
   }
 
+  // TODO: An error is raised here because a non-side-effect-free method is called above,
```

Comment:
Instead of `above`, state which method. Isn't `nextBoolean` side-effecting also? How will you soundly handle methods with side-effects?